qbft/docs/README.md

```diff
@@ -0,0 +1,91 @@
+- `ID`: an identifier similar to the one hold by the [`Controller`](#controller).
+- `Round`: the current round.
+- `Height`: the height of instance (that identifies the instance). In practice, this `Height` is the slot number of the duty.
+- `LastPreparedRound` and `LastPreparedValue`: the state regarding the last round in which the operator has prepared.
+- `ProposalAcceptedForCurrentRound`: the current round's proposal message, helpful to validate upcoming messages and store the proposed data.
+- `Decided` and `DecidedValue`: stores whether or not the instance has decided and, if so, for which value.
+- Four `MsgContainer` objects, one for each message type.
+
+## Instance
+
+`Instance` is an structure that represents a single QBFT execution. It holds a [`State`](#state), a [`IConfig`](#iconfig), and its start value (to be proposed if possible in case it's the leader).
+
+It has methods to start the the QBFT protocol and to process a message. Once processing a message, it also performs a message validation as described in the formal specification.
+
+The message processing function, which is called by the [`Controller`](#controller), returns the current state of whether it's decided or not (and the decided value) along with an aggregated commit message that will serve as the decided message.
+
+## Controller
+
+The main entry point of this module is the `Controller` structure. Its main goal is to allow the operator to manage sequential executions of the QBFT protocol (QBFT instances).
+
+The controller includes:
+- `Identifier`: to uniquely identify the QBFT controller. In this module, it's abstracted as a sequence of bytes but, in practice, it's a `types.MessageID` object that uniquely identifies a validator (or a committee for committee duties) and a duty type. This identifier is replicated into the [`Instance`](#instance)'s [`State`](#state) and into the QBFT messages.
+- `Height`: the current execution's height. Note that the QBFT executions are sequential so no execution run in parallel.
+- `StoredInstances`: a container to save the history of QBFT instances.
+- `CommitteeMember`: that contains information about the operator itself, such as its `OperatorID` and network public key, and about the whole committee of operators, including their `OperatorID`s, network public keys and the number of faulty nodes for such committee.
+- `config`: the [QBFT configuration](#iconfig) which is passed to its [`Instances`](#instance).
+
+
+The controller provides methods to start a new QBFT instance, provided of a given height and start value, and to process messages.
+
+If the received message is a Decided message, it simply updated the instance state's Commit container, if the instance exists. When processing a non-decided message, if the instance returns, for the first time, that it has decided, the controller broadcasts a Decided message (which is a quorum aggregation of Commit messages).
```

Comment:
Not sure what **Commit container** is, maybe it's **Decided container**:

simply updated the instance state's Commit container -> simply updated the instance state's Decided container